File: benchmarks/pandas/bench_series_format_table.py
"""
Benchmark: pandas Series.to_markdown() and Series.to_latex() on a 500-element Series.

Mirrors the tsb seriesToMarkdown and seriesToLaTeX benchmark.
Exercises table-rendering of both numeric and string series.
"""
import json
import time
import math
import pandas as pd

N = 500
WARMUP = 3
ITERATIONS = 30

num_data = [math.sin(i * 0.05) * 100 for i in range(N)]
str_data = [None if i % 10 == 0 else f"item_{i}" for i in range(N)]

num_series = pd.Series(num_data)
str_series = pd.Series(str_data)

# Warm-up
for _ in range(WARMUP):
    num_series.to_markdown()
    num_series.to_latex()
    str_series.to_markdown()
    str_series.to_latex()

start = time.perf_counter()
for _ in range(ITERATIONS):
    num_series.to_markdown()
    num_series.to_latex()
    str_series.to_markdown()
    str_series.to_latex()
total_ms = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "series_format_table",
    "mean_ms": total_ms / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total_ms,
}))

File: benchmarks/tsb/bench_series_format_table.ts
/**
 * Benchmark: seriesToMarkdown and seriesToLaTeX on a 500-element Series.
 *
 * Mirrors pandas Series.to_markdown() and Series.to_latex().
 * Exercises table-rendering of both numeric and mixed-type series.
 */
import { Series, seriesToMarkdown, seriesToLaTeX } from "../../src/index.ts";
import type { Scalar } from "../../src/types.ts";

const N = 500;
const WARMUP = 3;
const ITERATIONS = 30;

const numData: number[] = Array.from({ length: N }, (_, i) => Math.sin(i * 0.05) * 100);
const strData: Scalar[] = Array.from({ length: N }, (_, i) => (i % 10 === 0 ? null : `item_${i}`));

const numSeries = new Series({ data: numData });
const strSeries = new Series<Scalar>({ data: strData });

// Warm-up
for (let i = 0; i < WARMUP; i++) {
  seriesToMarkdown(numSeries);
  seriesToLaTeX(numSeries);
  seriesToMarkdown(strSeries);
  seriesToLaTeX(strSeries);
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  seriesToMarkdown(numSeries);
  seriesToLaTeX(numSeries);
  seriesToMarkdown(strSeries);
  seriesToLaTeX(strSeries);
}
const total_ms = performance.now() - start;

console.log(
  JSON.stringify({
    function: "series_format_table",
    mean_ms: total_ms / ITERATIONS,
    iterations: ITERATIONS,
    total_ms,
  }),
);
